cfg/excalibur/jec18ul/mc18_DYJets_madgraph/_make_configs.py

```diff
@@ -0,0 +1,110 @@
+
+    cfg = configtools.expand(cfg, ['basiccuts','finalcuts'], ['None', 'L1', 'L1L2L3'])
+
+    cfg['MPFSplittingJetPtMin'] = 15.
+    cfg['JNPFJetPtMin'] = 15.
+
+    # cfg['PileupWeightFile'] = os.path.join(configtools.getPath() , 'data/pileup/mc_weights/mc18ul_DYJets_madgraph_12Nov19/PUWeights_' + ''.join({runs}) + '_12Nov2019_DYJetsToLL_madgraphMLM.root')
+    cfg['PileupWeightFile'] = os.path.join(configtools.getPath() , 'data/pileup/jec18ul/PUWeights_2018_Data_ABCD_12Nov2019_UL2018_MC_Summer19-madgraphMLM_realistic_v11_L1v1.root')
+    cfg['NumberGeneratedEvents'] = 104017741
+    cfg['GeneratorWeight'] = 1.0
+    cfg['CrossSection'] = 6077.22  # from XSDB: https://cms-gen-dev.cern.ch/xsdb/?searchQuery=DAS=DYJetsToLL_M-50_TuneCP5_13TeV-madgraphMLM-pythia8
+
+    cfg['CutBackToBack'] = 0.44
+
+    cfg['VertexSummary'] = 'offlinePrimaryVerticesSummary'
+
+    cfg['ApplyElectronVID'] = True
+    cfg['ElectronVIDName'] = "Fall17-94X-V2"
+    cfg['ElectronVIDType'] = "cutbased"
+    cfg['ElectronVIDWorkingPoint'] = "tight"
+
+    if CH == 'ee':
+        cfg['Processors'].insert(cfg['Processors'].index('producer:ZJetValidElectronsProducer'), 'producer:ElectronCorrectionsProducer',)
+        cfg['ApplyElectronEnergyCorrections'] = True
+        cfg['ElectronEnergyCorrectionTags'] = ["electronCorrection:ecalTrkEnergyPostCorr"]
+    # TODO: Update to 2018
+    # elif CH == 'mm':
+    #     cfg['Processors'].insert(cfg['Processors'].index('producer:ValidMuonsProducer'), 'producer:MuonCorrectionsProducer',)
+    #     cfg['MuonRochesterCorrectionsFile'] = os.path.join(configtools.getPath(),'../Artus/KappaAnalysis/data/rochcorr/RoccoR2017UL.txt')
+    #     cfg['MuonEnergyCorrection'] = 'rochcorr2017ul'
+
+    cfg['CutJetID'] = 'tightlepveto'  # choose event-based JetID selection
+    cfg['CutJetIDVersion'] = '2018UL'  # for event-based JetID
+    cfg['CutJetIDFirstNJets'] = 2
+
+    cfg['Processors'] += ['producer:ZJetPUWeightProducer']
+    # cfg['ZJetPUWeightFiles'] = [os.path.join(configtools.getPath() ,'data/pileup/mc_weights/mc18ul_DYJets_madgraph_data_12Nov19/PUWeights_{{}}_12Nov2019_DYJetsToLL_madgraphMLM.root'.format(runperiod)) for runperiod in {runs}]
+    cfg['ZJetPUWeightFiles'] = [os.path.join(configtools.getPath() ,'data/pileup/jec18ul/PUWeights_2018_Data_{{}}_12Nov2019_UL2018_MC_Summer19-madgraphMLM_realistic_v11_L1v1.root'.format(runperiod)) for runperiod in {runs}]
+
+    cfg['ZJetPUWeightSuffixes'] = ['{{}}'.format(runperiod) for runperiod in {runs}]
+
+    cfg['Processors'] += ['producer:ZJetGenWeightProducer']
+    cfg['ZJetGenWeightNames'] = {lheWeightNames}
+
+    cfg['JetEtaPhiCleanerFile'] = os.path.join(configtools.getPath(), 'data/cleaning/jec18ul/Summer19UL18_V1/hotjets-UL18.root')
+    cfg['JetEtaPhiCleanerHistogramNames'] = ["h2hot_ul18_plus_hem1516_and_hbp2m1"]
+
+    cfg['HltPaths'] = {{
```

Comment:
It might be a good idea to leave these in though. If we decide to run the `HltFilter` or the jet eta-phi cleaning in MC in the future then we would just have to add the processors to the pipeline.

Reply:
Okay for me